### PR TITLE
issue-16 Add ArchUnit tests to the boilerplate in the upstream repo

### LINKED ISSUE #16: Add ArchUnit tests to this boilerplate as well
Would it be a good idea to bake in the ArchUnit tests into this boilerplate so it comes out of the box? Should I pick that up?

### PR DESCRIPTION
**Pull request by awkshwayrd**
Fixes #16 by adding maven dependency to xebia-archunit-extension

### CODE REVIEW

**Comment by shekhargulati** [comment]
@awkshwayrd you will also need to create settings.xml and change workflow as shown in this repo https://github.com/TobseF/HelloMaven/blob/master/.github/workflows/deploy.yml. Github package does not support unauthenticated access.

File: api/src/test/java/com/xebia/starters/api/ApiArchitectureTest.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+class ApiArchitectureTest extends AbstractArchitectureTests {
+
+
+    public ApiArchitectureTest() {
+        super(new ClassFileImporter()
+                        .withImportOption(ImportOption.Predefined.DO_NOT_INCLUDE_TESTS)
+                        .importPackagesOf(Application.class),
+                new String[]{
+                        Entity.class.getPackage().getName()
```
Comment (awkshwayrd) [inline api/src/test/java/com/xebia/starters/api/ApiArchitectureTest.java:17]: I feel like this way of passing in entity class package names is a bit awkward. I had to create a dummy class just to pass failing test. Should we tweak this in the archunit-extension somehow?

**Comment by awkshwayrd** [comment]
> @awkshwayrd you will also need to create settings.xml and change workflow as shown in this repo https://github.com/TobseF/HelloMaven/blob/master/.github/workflows/deploy.yml. Github package does not support unauthenticated access.

Let me work on this.

**Comment by awkshwayrd** [comment]
@shekhargulati The default token may not work here as we are trying to pull a 3rd party package rather than publish a package in the same repo. We may need a separate secret token to be created for this org with the correct scope and use it in the build action.

**Comment by awkshwayrd** [comment]
> [ERROR] Failed to execute goal on project api: Could not resolve dependencies for project com.xebia.starters:api:jar:0.1.0-SNAPSHOT: Failed to collect dependencies at com.github.xebia:archunit-extension:jar:0.2.0: Failed to read artifact descriptor for com.github.xebia:archunit-extension:jar:0.2.0: Could not transfer artifact com.github.xebia:archunit-extension:pom:0.2.0 from/to github (https://maven.pkg.github.com/xebia-os/xebia-archunit-extension): Authentication failed for https://maven.pkg.github.com/xebia-os/xebia-archunit-extension/com/github/xebia/archunit-extension/0.2.0/archunit-extension-0.2.0.pom 401 Unauthorized -> [Help 1]

**Comment by awkshwayrd** [comment]
Tried with a separate token GH_PACKAGES_TOKEN in my forked repo. It still didn't work.

**Comment by awkshwayrd** [comment]
The build passed in my fork after I hard-coded my Github username https://github.com/awkshwayrd/spring-boot-react-maven-starter/runs/1086893006

**Comment by awkshwayrd** [comment]
It also passed in my fork with hard-coded username and GITHUB_TOKEN. https://github.com/awkshwayrd/spring-boot-react-maven-starter/runs/1086915927
My feeling is GITHUB_USERNAME doesn't seem to be available.
It seems to have passed here as well, even with my username. Let me try removing username altogether.

File: api/src/test/java/com/xebia/starters/api/ApiArchitectureTest.java (hunk 1 above)
Comment (shekhargulati) [reply api/src/test/java/com/xebia/starters/api/ApiArchitectureTest.java:17]: I think we can make it optional. We can ignore the tests that don't have this set.

**Review by shekhargulati — APPROVED** [review]

**Pull request merged**